src/Functions/formatRow.cpp

```diff
@@ -38,8 +38,14 @@ class FunctionFormatRow : public IFunction
         : format_name(std::move(format_name_))
         , arguments_column_names(std::move(arguments_column_names_))
         , context(std::move(context_))
+        , format_settings(getFormatSettings(context))
     {
         FormatFactory::instance().checkFormatName(format_name);
+
+        /// We don't need handling exceptions while formatting as a row.
+        /// But it can be enabled in query sent via http interface.
+        format_settings.json.valid_output_on_exception = false;
+        format_settings.xml.valid_output_on_exception = false;
```

Comment:
The problem is that the `PeekableWriteBuffer` is used when these settings are enabled, but it works incorrectly with `WriteBufferFromVector` which is used in function `formatRow`. Probably it's worth to fix that or forbid using `WriteBufferFromVector` and maybe some other write buffers for using with `PeekableWriteBuffer`. However this change makes sense anyway because `PeekableWriteBuffer` adds an overhead of extra copying.